proxy/src/telemetry/metrics/latency.rs

```diff
@@ -146,18 +148,19 @@ where
 
 }
 
-
 impl<'a> IntoIterator for &'a Histogram {
     type Item = &'a u32;
```

Comment:
1. Why isn't `Item` be `Wrapping<u32>` or `prometheus::Counter`?

2. Why don't we just have a `pub fn buckets() -> &[Counter]` on `Histogram` and avoid the need to implement IntoIterator for it at all? A histogram itself isn't really an iterable thing, which is clear from the fact that its sum isn't included in the returned iterator, and `histogram.iter()` seems a lot less clear than `histogram.buckets().iter()`.

Reply:
I think having this implement `IntoIterator` rather than a function to borrow the bucket values slice is a relic of an earlier implementation, so that suggested change makes sense to me.

I suppose another option --- possibly a  better separation of concerns? --- would be to have `Histogram::into_iter` handle the joining of the bucket count values with the associated upper bounds, so the return type of the iterator was `(Latency, Counter)`? That behaviour is currently implemented in the `fmt::Display` impl, which seems like the wrong place for it...

Reply:
> suppose another option --- possibly a better separation of concerns? --- would be to have Histogram::into_iter handle the joining of the bucket count values with the associated upper bounds, so the return type of the iterator was (Latency, Counter)? That behaviour is currently implemented in the fmt::Display impl, which seems like the wrong place for it...

That makes sense to me, probably best done as a separate PR.

To answer my own question: The users of the `IntoIterator` implementation don't do arithmetic on the values, so having `Item` be a `Wrapping` type or anything else would just be annoying for them, without any gain.

Reply:
> The users of the IntoIterator implementation don't do arithmetic on the values, so having Item be a Wrapping type or anything else would just be annoying for them, without any gain.

Yeah, that's more or less the rationale behind the current code. Will probably revisit this in an additional PR, as you requested. 

Note also that some of the ugliness here is because the `latency` module's API is consumed by both `prometheus` _and_ the push metrics reporter, which --- although slated for removal --- is still in the codebase. When that goes away and `latency` only has to support the prometheus server, this code will be much easier to refactor.